android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java

```diff
@@ -127,25 +126,27 @@ public void handleIntent(Intent intent) {
         searchQuery(query);
     }
 
+    @Override
+    public void onSaveInstanceState(Bundle bundle) {
+        if (searchView != null && searchViewModel != null) {
+            searchViewModel.setSearchText(searchText);
+        }
+        super.onSaveInstanceState(bundle);
+    }
+
     private void searchQuery(String constraint) {
         results.clear();
+        globalSearchAdapter.clear();
 
         if (!TextUtils.isEmpty(constraint)) {
-            String query = constraint.toLowerCase(Locale.getDefault());
-            String wildcardQuery = String.format("*%s*", query);
-            addResultsFromTracks(wildcardQuery);
-            addResultFromSessions(wildcardQuery);
-            addResultsFromSpeakers(wildcardQuery);
-            addResultsFromLocations(wildcardQuery);
-        }
-    }
-
-    @Override
-    protected void onSaveInstanceState(Bundle outState) {
-        if (searchView != null) {
-            outState.putString(SEARCH, searchText);
+            searchViewModel.getSearchResults().observe(SearchActivity.this, searches -> {
+                results.add(searches);
+                globalSearchAdapter.setCopyOfSearches(searches);
+                globalSearchAdapter.notifyDataSetChanged();
+                handleVisibility();
+            });
```

Comment:
@srv-twry I am keeping all the results in the viewmodel. I am fetching those in the above `searchQuery` method

Reply:
I asked about the searchText variable that you've stored in the ViewModel, where are you using the ```getSearchText``` method of the ViewModel inside the Activity ?

Reply:
Oh,I completely forgot to invoke the function in the `searchQuery` method.